packages/graphql-server/src/resolvers/group.ts

```diff
@@ -269,6 +272,56 @@ export const update = async (root, args, context: Context) => {
   return transform(group);
 };
 
+export const onPhAppDeleted = async (context: Context, phApplication: PhApplication) => {
+  const {kcAdminClient} = context;
+
+  // update resource
+  const group = await findGroupByName(phApplication.groupName, kcAdminClient);
+
+  // merge attrs
+  // displayName, canUseGpu, quotaGpu in attributes
+  const attrs = new Attributes({
+    keycloakAttr: group.attributes,
+    schema: attrSchema
+  });
+  const mlflowTrackingUri = getFromAttr('mlflow-tracking-uri', group.attributes, null);
+  const matched = mlflowTrackingUri.match(/^http:\/\/app-(?<app>mlflow-.+):5000$/);
```

Comment:
b/c looks like when user has different apps, your current function still remove setting of the group by delete any app even not mlflow app.

Reply:
The reason is that we cannot gruantee that the appName must be `mlflow`. It can be `mlflow-1.8' for customize mlflow app.

And because this is mlflow setting, if the app is deleted, even it is not original mlflow app template, it still make sense to remove the setting because the app does not exist anymore.

Reply:
Fixed
image: sc20238-0d3f7c0